### ISSUE TITLE
Embedding images with relative paths

### ISSUE THREAD

**Issue by bcmadsen** [RickStrahl/MarkdownMonster#681]
I am trying to embed an image. I notice that this doesn't actually embed the image in the preview window:
```text
![](image.jpg)
```
&hellip; although this does:
```text
![](.\image.jpg)
```
&hellip; and so does:
```text
![](.\\image.jpg)
```
Moving up to a parent folder, this also works:
```text
![](..\image.jpg)
```
&hellip; and this too:
```text
![](..\\image.jpg)
```
&hellip; but in the grandparent folder, where the relative path includes two (or more) levels of double-dots, the image isn't embedded anymore. I've tried both:
```text
![](..\..\image.jpg)
```
and:
```text
![](..\\..\\image.jpg)
```
Needless to say, the image exists at all of these locations. But Markdown Monster isn't embedding the image in its preview window in the first case, nor in either of the last two.

Am I expecting something I shouldn't expect?

**Comment by RickStrahl** [comment]
No that should work just fine. Unpathed image links aren't fixed up in any way. Especially a local path should always work unless you're maybe not in the right folder and the image isn't actually there?

In the Folder Browser click on the arrow icon to navigate to that folder and see if the image is actually there, and click on it to see if the image displays. Is it possible the image is corrupt and just not rendering in the preview? Also can you try Shift-F12 to preview in browser and see if the image shows there and then check the HTML with the inspector to see what actually got embedded into the document?

MM does some path fix ups for **some things**, but only paths that start with a `/`. FWIW paths in the editor should be **Web paths** not OS paths (ie. forward slashes). The preview will render it properly I think but actual browsers may not.

**Comment by bcmadsen** [comment]
I think the thing I was expecting that I shouldn't have expected was the use of backslashes rather than forward slashes. This works:
```text
![](../../image.jpg)
```
&hellip; but these don't:
```text
![](..\..\image.jpg)
![](..\\..\\image.jpg)
```

So as long as I keep in mind that I need to use forward slashes for these things, I think I'll be all right.

**Comment by RickStrahl** [comment]
Well MM doesn't fix up the URLs unless it has to (ie. for fixed base paths) so what you see is what you get. Browsers don't really work with backslashes although IE (which is what the previewer uses) should actually work with that. Especially since the actual preview image is rendered from a file on disk with a `file:///` path.

But yeah, you should definitely use forward slashes always.

But you mentioned that hte image in the same folder (without a path) didn't work. That should **always** work no matter what.

**Comment by bcmadsen** [comment]
I followed your suggestion about making sure MM's default folder was the same as the folder where my document was, saw that I wasn't, so starting clicking some things until I landed in the right place, and now that's working like I expect. Weirdly, I can't make it fail anymore. I don't know what I was experiencing in the situation I was in, and since I can't reproduce it, now I'm reluctant to continue reporting it.

As part of that whole exercise, I notice that the "Folder Path" setting isn't changing along with the files I'm opening. I have never really understood why that doesn't ever seem to change to the folder where the files I'm actively looking at live, even though the text below it asserts that it reflects "the last file that was active". However, MM is behaving normally in pretty much every other way, so that doesn't really cause me much concern, other than puzzlement.

Bottom line: yeah, I know. The `![](image.jpg)` construction is working now, and I can't explain why it wasn't before. If I figure it out later on, I'll return and report.

**Comment by RickStrahl** [comment]
The Folder browser **should most definitely** not navigate when you change document location. No editor does this - take a look at VS code or Visual Studio or any editor. While it sounds good on paper in reality you typically want to see a fixed location in which you browse around as needed.

There are short cuts to open the current document folder (right click the tab, the navigation icon in the folder browser, and I just added a menu option on the Window menu (alt-w-b).
